### ISSUE TITLE
Makefile improvement suggestions

### ISSUE THREAD

**Issue by ncdc** [kubernetes-sigs/cluster-api-provider-aws#776]
/kind feature

**Describe the solution you'd like**
I'd like to propose some changes to our Makefile targets:

- If I'm compiling locally, I only want the clientsets/listers/informers regenerated as-needed; in other words, do not regenerate every time I run any sort of build-related target. The same applies to the `go generate` output.
- If I'm compiling locally, I only want to deal with `dep` as-needed; in other words, do not run `dep ensure` unless I explicitly ask for it.

tl;dr: try to speed up builds by switching targets that aren't needed frequently to optional/opt-in instead of as dependencies

**Anything else you would like to add:**
N/A

**Comment by davidewatson** [comment]
+1

Usually, regeneration is unnecessary. And when it is it should be caught during CI and added as a separate (intentional) commit.

**Comment by ncdc** [comment]
We've addressed a number of my complaints during the v1alpha2 development cycle.
/close
